### PR TITLE
Private and protected properties and methods

### PR DESCRIPTION
**Pull request by mahdiHash**
Please review and comment/commit if anything is wrong

### CODE REVIEW

File: 1-js/09-classes/04-private-protected-properties-methods/article.md [hunk 1]
```diff
@@ -1,95 +1,95 @@
 
-# Private and protected properties and methods
+# ویژگی‌ها و متدهای شخصی و محافظت شده
 
-One of the most important principles of object oriented programming -- delimiting internal interface from the external one.
+یکی از مهم‌ترین قواعد برنامه‌نویسی شیءگرا -- محدود کردن رابط داخلی از رابط بیرونی است.
 
-That is "a must" practice in developing anything more complex than a "hello world" app.
+یعنی یک کارِ «بایدی» در توسعه هر چیزی پیچیده‌تر از یک برنامه "hello world".
```
Comment (mahdiHash) [inline 1-js/09-classes/04-private-protected-properties-methods/article.md:6]: This line needs more attention.

File: 1-js/09-classes/04-private-protected-properties-methods/article.md [hunk 2]
```diff
@@ -1,95 +1,95 @@
...
-In object-oriented programming, properties and methods are split into two groups:
+در برنامه‌نویسی شیءگرا، ویژگی‌ها و متدها به دو گروه تقسیم می‌شوند:
 
-- *Internal interface* -- methods and properties, accessible from other methods of the class, but not from the outside.
-- *External interface* -- methods and properties, accessible also from outside the class.
+- *رابط درونی* -- متدها و ویژگی‌ها، قابل دسترس از متدهای دیگر کلاس، اما نه از بیرون.
+- *رابط بیرونی* -- متدها و ویژگی‌ها، قابل دسترس از بیرون از کلاس.
 
-If we continue the analogy with the coffee machine -- what's hidden inside: a boiler tube, heating element, and so on -- is its internal interface.
+اگر ما مثال قهوه‌ساز را ادامه دهیم -- چیزی که درون آن است: یک مجرای بخار، المنت حرارت و غیره -- رابط درونی است.
```
Comment (mahdiHash) [inline 1-js/09-classes/04-private-protected-properties-methods/article.md:41]: This line needs more attention.

File: 1-js/09-classes/04-private-protected-properties-methods/article.md [hunk 3]
```diff
@@ -112,22 +112,22 @@ class CoffeeMachine {
...
+حالا دسترسی تحت کنترل است پس تنظیم مقدار آب کمتر از صفر ممکن نیست.
 
-## Read-only "power"
+## ویژگی "power" فقط‌خواندنی
 
-For `power` property, let's make it read-only. It sometimes happens that a property must be set at creation time only, and then never modified.
+بیایید ویژگی `power` را فقط‌خواندنی کنیم. گاهی اوقات یک ویژگی باید فقط زمان ایجاد کردن مقداردهی شود و دیگر هیچ‌وقت تغییر نکند.
 
-That's exactly the case for a coffee machine: power never changes.
+این دقیقا برای قهوه‌ساز هم صدق می‌کند: توان (power) هیچ‌وقت تغییر نمی‌کند.
```
Comment (mahdiHash) [inline 1-js/09-classes/04-private-protected-properties-methods/article.md:128]: This line needs more attention.

File: 1-js/09-classes/04-private-protected-properties-methods/article.md [hunk 4]
```diff
@@ -280,43 +280,43 @@ class User {
...
-Supportable
-: The situation in programming is more complex than with a real-life coffee machine, because we don't just buy it once. The code constantly undergoes development and improvement.
+قابل پشتیبانی
+: این وضعیت در برنامه‌نویسی پیچیده‌تر از یک قهوه‌ساز در زندگی واقعی است چون ما فقط آن را یک بار نمی‌خریم. کد دائما در توسعه و پیشرفت است.
 
-    **If we strictly delimit the internal interface, then the developer of the class can freely change its internal properties and methods, even without informing the users.**
+    **اگر ما به صورت سخت‌گیرانه رابط درونی را جداسازی کنیم، سپس توسعه‌دهنده‌ی کلاس آزادانه می‌تواند ویژگی‌ها و متدهای درون آن را تغییر دهد، حتی بدون اینکه کاربران را خبردار کند.**
 
-    If you're a developer of such class, it's great to know that private methods can be safely renamed, their parameters can be changed, and even removed, because no external code depends on them.
+    اگر شما توسعه‌دهنده چنین کلاسی باشید، این عالی است که بتوانید متدهای خصوصی را با خیال راحت تغییر نام دهید، پارامترهای آن‌ها را تغییر دهید و حتی حذف کنید، چون هیچ کد بیرونی به آن‌ها وابسته نیست.
```
Comment (mahdiHash) [inline 1-js/09-classes/04-private-protected-properties-methods/article.md:306]: This line needs more attention.

File: 1-js/09-classes/04-private-protected-properties-methods/article.md [hunk 5]
```diff
@@ -280,43 +280,43 @@ class User {
...
 
-    For users, when a new version comes out, it may be a total overhaul internally, but still simple to upgrade if the external interface is the same.
+    برای کاربران، زمانی که نسخه‌ای جدید منتشر می‌شود، ممکن است از درون تعمیرات اساسی نیاز داشته باشد اما اگر رابط بیرونی یکسان باشد هنوز برای ارتقا دادن ساده است.
```
Comment (mahdiHash) [inline 1-js/09-classes/04-private-protected-properties-methods/article.md:308]: This line needs more attention.

File: 1-js/09-classes/04-private-protected-properties-methods/article.md [hunk 6]
```diff
@@ -280,43 +280,43 @@ class User {
...
-To hide an internal interface we use either protected or private properties:
+برای پنهان‌سازی یک رابط درونی می‌توانیم یا از ویژگی‌های محافظت‌شده استفاده کنیم یا ویژگی‌های خصوصی:
 
-- Protected fields start with `_`. That's a well-known convention, not enforced at the language level. Programmers should only access a field starting with `_` from its class and classes inheriting from it.
-- Private fields start with `#`. JavaScript makes sure we can only access those from inside the class.
+- فیلدهای محافظت‌شده با `_` شروع می‌شوند. این یک قرارداد شناخته‌شده است و در سطح زبان اجرایی نشده‌اند. برنامه‌نویسان فقط باید از درون کلاس و کلاس‌هایی که از آن ارث می‌برند، به فیلدی که با `_` شروع می‌شود دسترسی پیدا کنند.
+- فیلدهای خصوصی با `#` شروع می‌شوند. جاوااسکریپت مطمئن می‌شود که ما فقط از درون کلاس بتوانیم به آن‌ها دسترسی پیدا کنیم.
 
-Right now, private fields are not well-supported among browsers, but can be polyfilled.
+در حال حاضر، فیلدهای خصوصی به خوبی در میان مرورگرها پشتیبانی نمی‌شوند اما می‌توان برای آن‌ها پلیفیل ساخت.
```
Comment (mahdiHash) [inline 1-js/09-classes/04-private-protected-properties-methods/article.md:322]: This line needs more attention.

File: 1-js/09-classes/04-private-protected-properties-methods/article.md (hunk 1 above)
Comment (mahdyar) [reply 1-js/09-classes/04-private-protected-properties-methods/article.md:6]:
```suggestion
یعنی یک کارِ «بایدی» در توسعه هر چیزی پیچیده‌تر از یک برنامه‌ی «hello world».
```

File: 1-js/09-classes/04-private-protected-properties-methods/article.md (hunk 2 above)
Comment (mahdyar) [reply 1-js/09-classes/04-private-protected-properties-methods/article.md:41]:
```suggestion
اگر ما مقایسه را با قهوه‌ساز ادامه دهیم -- چیزی که درون آن است: یک مجرای بخار، المنت حرارت و غیره -- رابط درونی است.
```

File: 1-js/09-classes/04-private-protected-properties-methods/article.md (hunk 4 above)
Comment (mahdyar) [reply 1-js/09-classes/04-private-protected-properties-methods/article.md:306]:
```suggestion
    اگر شما توسعه‌دهنده چنین کلاسی باشید، این خوب است که بتوانید متدهای خصوصی را با خیال راحت تغییر نام دهید، پارامترهای آن‌ها را تغییر دهید و حتی حذف کنید، چون هیچ کد بیرونی به آن‌ها وابسته نیست.
```

File: 1-js/09-classes/04-private-protected-properties-methods/article.md (hunk 6 above)
Comment (mahdyar) [reply 1-js/09-classes/04-private-protected-properties-methods/article.md:322]:
```suggestion
در حال حاضر، فیلدهای خصوصی به خوبی در میان مرورگرها پشتیبانی نمی‌شوند اما می‌توان برای آن‌ها از پلیفیل‌ها استفاده کرد.
```

**Review by mahdyar — CHANGES_REQUESTED** [review]
Please review the requested changes.

**Comment by mahdiHash** [comment]
Thank you @mahdyar 
/done

**Review by mahdyar — APPROVED** [review]
Thank you, @mahdiHash!

**Pull request merged**